pkg/binding/interfaces.go

```diff
@@ -98,6 +108,10 @@ type StructuredEncoder interface {
 // Protocols that supports binary encoding should implement this interface to implement direct
 // binary -> binary transfer.
 type BinaryEncoder interface {
+	//TODO explain init and end with relative propagation!!!
+
+	Init() error
```

Comment:
`Init` implies only once per `BinaryEncoder` lifecycle, and guessing from usage, this happens once per message encoding? Please explain.

Reply:
If once per BinaryEncoder, then init is correct. But if once per message, then something like `Start` or `Begin` might make more sense

Reply:
So `BinaryEncoder` interface is usually tied to one single "output" message, it's not reused (https://github.com/cloudevents/sdk-go/pull/302/files#diff-c5d6e121af538561803b092335cebcefR45)

`Init()` is run at the beginning of the encoding, so yeah maybe `Start()` is better